src/java/org/apache/cassandra/serializers/ListSerializer.java

```diff
@@ -150,18 +150,11 @@ public <V> boolean anyMatch(V input, ValueAccessor<V> accessor, Predicate<V> pre
 
             for (int i = 0; i < s; i++)
             {
-                int size = accessor.getInt(input, offset);
-                if (size < 0)
-                    continue;
-
-                offset += TypeSizes.INT_SIZE;
-
-                V value = accessor.slice(input, offset, size);
+                V value = readValue(input, accessor, offset);
+                offset += sizeOfValue(value, accessor);
```

Comment:
Why did you move this line up? It will be not needed in case of the predicate returning true

Reply:
I do no remember exactly but I think that some of the changes I did started to use that code in a different way and triggered an exception as the code was not moving the offset.